internal/target/service_list.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package target
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/refreshtoken"
+	"github.com/hashicorp/boundary/internal/types/resource"
+)
+
+func List(
+	ctx context.Context,
+	repo *Repository,
+	grantsHash []byte,
+	pageSize int,
+	filterItemFn func(Target) (bool, error),
```

Comment:
It might be a good idea to define this function as a type in the target package, ie:

```go
type ListFilterFunc func(Target) (bool, error)
```

Then the type can be documented to explain how it should behave.